app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+
+    for person_data in people:
+        person_list = [Person(person_data["name"],
+                              person_data["age"]) for person_data in people]
+
+    for person_data in people:
```

Comment:
Doing two for loops and using 4 conditional statements is not efficient. Your logic should be like so:
1. Iterate through data with people info.
2. Create a Person on next iteration.
3. if "wife" in data and a Person with wife name already in Person.people list -> set wife and husband attributes.
4. Same check for "husband"
5.  Add Person to the list.
6. return list